spirv_glsl.cpp

```diff
@@ -3535,70 +3559,53 @@ void CompilerGLSL::emit_extension_workarounds(spv::ExecutionModel model)
 				switch (e)
 				{
 				case Supp::NV_shader_thread_group:
-					statement("bool subgroupElect() {");
-					statement("int firstLive = findLSB(ballotThreadNV(true));");
-					statement("return gl_ThreadInWarpNV == uint(firstLive);");
-					statement("}");
+					statement("uvec4 subgroupBallot(bool v) { return uvec4(ballotThreadNV(v), 0u, 0u, 0u); }");
+				case Supp::ARB_shader_ballot:
+					statement("uvec4 subgroupBallot(bool v) { return uvec4(unpackUint2x32(ballotARB(v)), 0u, 0u); }");
+					break;
+				default:
 					break;
 				}
 			}
-			statement("#else");
+			statement("#endif");
+		}
+		if (shader_subgroup_supporter.is_feature_requested(Supp::SubgroupElect))
+		{
+			statement("#ifndef GL_KHR_shader_subgroup_basic");
 			statement("bool subgroupElect() {");
-			statement("uvec2 activeMask = unpackUint2x32(ballotARB(true));");
-			statement("int firstLive = findLSB(activeMask.x);");
-			statement("if (firstLive == -1)");
-			statement("firstLive = findLSB(activeMask.y)+32;");
-			statement("return gl_SubgroupInvocationID == uint(firstLive);");
+			statement("uvec4 activeMask = subgroupBallot(true);");
+			statement("uint firstLive = subgroupBallotFindLSB(activeMask);");
+			statement("return gl_SubgroupInvocationID == firstLive;");
```

Comment:
does NV/AMD not have specific instructions for electing an invocation?

Reply:
@Crisspl ?

Reply:
Looked for in https://www.khronos.org/registry/OpenGL/extensions/ , havent found anything directly corresponding